WinUIGallery/TabViewPages/TabViewWindowingSamplePage.xaml.cs

```diff
@@ -69,108 +63,67 @@ public void AddTabToTabs(TabViewItem tab)
             Tabs.TabItems.Add(tab);
         }
 
-        // Create a new Window once the Tab is dragged outside.
-        private void Tabs_TabDroppedOutside(TabView sender, TabViewTabDroppedOutsideEventArgs args)
+        private void Tabs_TabTearOutWindowRequested(TabView sender, TabViewTabTearOutWindowRequestedEventArgs args)
         {
             var newPage = new TabViewWindowingSamplePage();
 
-            Tabs.TabItems.Remove(args.Tab);
-            newPage.AddTabToTabs(args.Tab);
+            tabTearOutWindow = WindowHelper.CreateWindow();
+            tabTearOutWindow.ExtendsContentIntoTitleBar = true;
+            tabTearOutWindow.Content = newPage;
+            tabTearOutWindow.AppWindow.SetIcon("Assets/Tiles/GalleryIcon.ico");
+            newPage.SetupWindowMinSize(tabTearOutWindow);
 
-            var newWindow = WindowHelper.CreateWindow();
-            newWindow.ExtendsContentIntoTitleBar = true;
-            newWindow.Content = newPage;
-            newPage.SetupWindowMinSize(newWindow);
-
-            newWindow.Activate();
+            args.NewWindowId = tabTearOutWindow.AppWindow.Id;
         }
 
-        private void Tabs_TabDragStarting(TabView sender, TabViewTabDragStartingEventArgs args)
+        private void Tabs_TabTearOutRequested(TabView sender, TabViewTabTearOutRequestedEventArgs args)
         {
-            // We can only drag one tab at a time, so grab the first one...
-            var firstItem = args.Tab;
+            if (tabTearOutWindow == null)
+            {
+                return;
+            }
 
-            // ... set the drag data to the tab...
-            args.Data.Properties.Add(DataIdentifier, firstItem);
+            var newPage = (TabViewWindowingSamplePage)tabTearOutWindow.Content;
 
-            // ... and indicate that we can move it
-            args.Data.RequestedOperation = DataPackageOperation.Move;
+            foreach (TabViewItem tab in args.Tabs.Cast<TabViewItem>())
+            {
+                GetParentTabView(tab)?.TabItems.Remove(tab);
+                newPage.AddTabToTabs(tab);
+            }
         }
 
-        private async void Tabs_TabStripDrop(object sender, DragEventArgs e)
+        private void Tabs_ExternalTornOutTabsDropping(TabView sender, TabViewExternalTornOutTabsDroppingEventArgs args)
         {
-            // This event is called when we're dragging between different TabViews
-            // It is responsible for handling the drop of the item into the second TabView
+            args.AllowDrop = true;
+        }
+
+        private void Tabs_ExternalTornOutTabsDropped(TabView sender, TabViewExternalTornOutTabsDroppedEventArgs args)

```

Comment:
@llongley it's a bit odd that it's a collection of UIElement when you'd only be dragging TabViewItems, also the base control doesn't support selecting multiple tabs currently.

Also, could these event args just provide the external parent TabView of the external TabViewItem so the VisualTree isn't needed here? Wouldn't it just be passing an additional reference to `this` as the event creating this is the Drag event on that TabView/Window pair?

https://github.com/microsoft/microsoft-ui-xaml/blob/f31293f4227cd3b4f7c151106528b8e1367bda11/src/controls/dev/TabView/TabView.cpp#L1902-L1935